src/layout/NestedPicker.js

```diff
@@ -0,0 +1,120 @@
+import React from 'react'
+import F from 'futil-js'
+import _ from 'lodash/fp'
+import { inject, observer } from 'mobx-react'
+import { observable } from 'mobx'
+import { withStateLens } from '../utils/mobx-react-utils'
+import TextHighlight from './TextHighlight'
+
+// Unflatten by with support for arrays (allow dots in paths) and not needing a _.keyBy first
+let unflattenObjectBy = _.curry((iteratee, x) =>
+  _.zipObjectDeep(F.mapIndexed(iteratee, x), _.values(x))
+)
+
+let isField = x => x.typeDefault
+
+let FilteredSection = observer(
+  ({ options, onClick, highlight, Highlight, Item }) => (
+    <div>
+      {F.mapIndexed(
+        (option, field) => (
+          <Item key={field} onClick={() => onClick(option.value)}>
+            <Highlight text={option.label} pattern={highlight} />
+          </Item>
+        ),
+        options
+      )}
+    </div>
+  )
+)
+let Section = observer(({ options, onClick, selected, Item }) => (
+  <div>
+    {F.mapIndexed(
+      (item, key) => (
+        <Item
+          key={key}
+          onClick={() => onClick(item.value || key, item)}
+          active={selected === key}
+          disabled={selected && selected !== key}
+          hasChildren={!isField(item)}
+        >
+          {isField(item) ? item.shortLabel || item.label : _.startCase(key)}
+        </Item>
+      ),
+      options
+    )}
+  </div>
+))
+
+let toNested = _.flow(
+  _.map(x => _.defaults({ path: x.value }, x)),
+  unflattenObjectBy('path')
+)
+let PanelTreePicker = inject((store, { onChange, options }) => {
+  let x = {
+    state: observable({ selected: [] }),
+    nestedOptions: toNested(options),
+    select: level => (key, field) => {
```

Comment:
Would it be better to pass `level` as a property of an object? so it would be more clear when you call it `select({ level: 0 })`. Or perhaps rename `select` to be something like `selectLevel`.

Reply:
The represents selecting a field at a certain level, but agree that the nature of the currying isn't as clear as it could be 🤔

Reply:
It's not the currying, it's she 'select', since we're also talking about a nested `picker`. Select in my head is about picking an element from a list, not between levels.